RFCs/0035-partiql-window-function.md

```diff
@@ -0,0 +1,1224 @@
+
+Notice that to demonstrate the difference visually, the result will be sorted in descending order.
+
+`<func> OVER (PARTITION BY ticker ORDER BY price DESC)  -- has PARTITION BY and ORDER BY sub-clause`
+
+![Example 1.1.5 Graph](./0035-partiql-window-function/Example_1_4.png)
+
+```
+<<
+ -- Row 1
+  {
+    'trade_date' : 2022-09-30,
+    'ticker' : 'AMZN', 
+    'price', 113.00
+    -- partition for row 1
+    'partition' : 
+      [
+          <<
+            'trade_date' : 2022-10-04,
+            'ticker' : 'AMZN', 
+            'price' : 121.09
+          >>,
+          <<
+            'trade_date' : 2022-10-03,
+            'ticker' : 'AMZN', 
+            'price' : 115.88
+          >>,
+          <<
+            'trade_date' : 2022-09-30,
+            'ticker' : 'AMZN', 
+            'price' : 113.00
+          >>
+      ]
+  }
+  -- rest of the data skiped
+  ...  
+>>
+```
+
+#### WINDOWED Clause 
+
+```sql
+WINDOWED (
+    PARTITION BY e1, e2, ..., em
+    ORDER BY (o1 [ASC|DESC]? [NULLS FIRST| NULLS LAST]?,
+              ...,
+              on [ASC|DESC]? [NULLS FIRST| NULLS LAST]?
+    ) PARTITION AS p AT pos
+```
+
+Where $e_1,...,e_n$ is a list of **partition expressions**, $o_1,....,o_n$ is a list of **ordering expressions**, $p$ is the **partition variable**, and $pos$ is the **position variable** which indicates the position of the current row in the partition.
+
+The `WINDOWED` clause can be viewed as a modular function, with $B^{in}_{WINDOWED}$ (a bag of binding tuples) as an input parameter to the function.
+
+
+1. PARTITION BY sub-clause:
+
+* If PARTITION BY is presented: $B^{in}_{WINDOWED}$ is partitioned into the minimal number of equivalence partition $B_1,...,B_n$. The equivalence rule is the same as the one used in `GROUP BY` clause (Appendix 2). 
+* If there is no `PARTITION BY`, the entire input binding collection is considered as one partition.
+* Each of the $B_i$ produced by `PARTITION BY` SHALL be a list instead of a bag, even though the order is non-deterministic at the moment. The partition needs to be a list because we want to use the index to locate the original row.
+
+2. ORDER BY sub-clause:
+
+* If ORDER BY is presented, each equivalence partition is ordered based on the rules defined in Spec Section 12 (Appendix 3).
+* Notice that the ORDER BY in WINDOWED clause applies to partition only, it does not necessarily affect the final order of the result.
+
+3. WINDOWED clause output:
+
+* For each binding tuple $b_i$ in $B^{in}_{WINDOWED}$, output $b = b_i || < p : B_i, pos: x>$ where $b_i$ in $B_i$ and $p[x] = b_i$.
+* Notice that in case of duplication, the pos is nondeterministic but **MUST** be guaranteed to be unique.
+
+Logically, `WINDOWED` clause is evaluated after `GROUP BY` and `HAVING` and before `ORDER BY`, the evaluation order looks like:
+
+```
+FROM -> WHERE -> GROUP BY -> HAVING -> WINDOWED -> ORDER BY -> LIMIT -> SELECT 
+```
+
+#### Examples
+
+Suppose  $B^{in}_{WINDOWED}$ is:
```

Comment:
I really like how spec — when exemplifying — starts from the actual data and also the query and then steps through each stage with Bin and Bouts. Can we also do the same here for at least one of the examples?

Reply:
Addressed in the new commit. See example 4.1 (sorry changed the index system a bit)